### PR TITLE
Update some of our readme files

### PR DESCRIPTION
**Pull request by cyli**
- Added new features in master that are not in 0.4.2 to the changelog for the next release
- Updated the README and CONTRIBUTING doc to reflect the fact that master is our development branch.

### CODE REVIEW

**Comment by cyli** [comment]
@endophage Good idea, updated the `CONTRIBUTING.md` doc to specifically mention it.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,11 +1,21 @@
 # Changelog
 
-## [v0.4.1](https://github.com/docker/notary/releases/tag/v0.4.0) 9/27/2016
+## v0.5-dev (in progress)
++ Non-certificate public keys in PEM format can now be added to delegation roles [#965](https://github.com/docker/notary/pull/965)
++ PostgresSQL support as a storage backend for Server and Signer [#920](https://github.com/docker/notary/pull/920)
++ Notary server's health check now fails if it cannot connect to the signer, since no new repositories can be created and existing repositories cannot be updated if the server cannot reach the signer [#952](https://github.com/docker/notary/pull/952)
++ Server runs its connectivity healthcheck to the server once every 10 seconds instead of once every minute. [#902](https://github.com/docker/notary/pull/902)
++ The keys on disk are now stored in the `~/.notary/private` directory, rather than in a key hierarchy that separates them by GUN and by role.  Notary will automatically migrate old-style directory layouts to the new style.  **This is not forwards-compatible against notary<0.4.1 and docker<=1.12** [#872](https://github.com/docker/notary/pull/872)
```
Comment (riyazdf) [inline CHANGELOG.md]: nit: `notary<0.4.2`?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,11 +1,21 @@
 # Changelog
 
-## [v0.4.1](https://github.com/docker/notary/releases/tag/v0.4.0) 9/27/2016
+## v0.5-dev (in progress)
++ Non-certificate public keys in PEM format can now be added to delegation roles [#965](https://github.com/docker/notary/pull/965)
++ PostgresSQL support as a storage backend for Server and Signer [#920](https://github.com/docker/notary/pull/920)
```
Comment (riyazdf) [inline CHANGELOG.md]: nit: `PostgreSQL`

Comment (cyli) [inline CHANGELOG.md]: Yes, thanks!

File: CHANGELOG.md (hunk 1 above)
Comment (cyli) [inline CHANGELOG.md]: Hmm... 0.4.1 does have the relevant change, and it is tagged.  But it is also not released.  I think 0.4.1 is technically more correct, but 0.4.2 is more friendly.  Will be happy to make the change if you feel it'd be better

Comment (cyli) [inline CHANGELOG.md]: Meh, nm, I'm just going to make the change.  They should just probably be using 0.4.2.

Comment (riyazdf) [inline CHANGELOG.md]: thanks! I agree that 0.4.2 might be a little easier to understand since it's our latest release.

**Pull request merged**